scripts/build/debug/300-gdb.sh

```diff
@@ -69,11 +69,8 @@ do_debug_gdb_build() {
 
         cross_extra_config=("${extra_config[@]}")
         cross_extra_config+=("--with-expat")
-        # NOTE: DO NOT USE --with-libexpat-prefix (until GDB configure is smarter)!!!
```

Comment:
> Isn't absence of --with-libexpat-prefix equivalent to --without-libexpat-prefix?
No, they are not equivalent at all! This is the point of the commit and PR.


> What is the change in behaviour between these two configurations?
--with-expat-prefix=<folder> just adds an additional folder to search for expat, it does not prevent peeking into $prefix/lib
--without-expat-prefix forbids looking to $prefix/lib.


> Even if this change is needed, I suggest to keep the comment (since your --without-libexpat-prefix only reinforces what this comment states - that --with-libexpat-prefix is harmful).
I think this comment is misleading.
GDB configure (more precisely, AC_LIB_HAVE_LINKFLAGS) is smart, but extremely "unfriendly": it is not trivial to come to "--without-expat-prefix" to achieve our goal. The real problem is that AC_LIB_HAVE_LINKFLAGS is not properly documented for cross-compiling with DESTDIR.